packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.ts

```diff
@@ -261,6 +360,91 @@ export default createESLintRule<Options, MessageIds>({
             data: {
               imp,
             },
+            fix(fixer) {
+              // imp is equal to @myorg/someproject
+              const indexTsPaths = getBarrelEntryPointByImportScope(imp);
+              if (indexTsPaths && indexTsPaths.length > 0) {
+                // imported JS functions to remap
+                const imports = (
+                  (node.source.parent as any).specifiers || []
+                ).map((specifier) => specifier.imported.name);
+
+                // ignore non-named imports for now, like "import '../../some-other-lib/src'"
```

Comment:
This should have been picked up above where we check if it's an import into another package.
Additionally, the example shows relative import, while the condition on line 354 explicitly checks against those.

What is the purpose of this early bail? Can we document this in the E2E test or unit test?

Reply:
Please leave a TODO as a reminder